dynamodb/README.md

```diff
@@ -0,0 +1,134 @@
+certain amount of sampled data, by default, 100000 rows. If a table has more
+rows than the default value, we only use 100000 rows for estimating the schema.
+You can change this value, run
+
+```sh
+harbourbridge -driver=dynamodb -schema-sample-size=1000000
+```
+
+## Schema Conversion
+
+The HarbourBridge tool maps DynamoDB types to Spanner types as follows:
+
+| PostgreSQL Type    | Spanner Type               | Notes                                     |
+| ------------------ | -------------------------- | ----------------------------------------- |
+| `Number`           | `NUMERIC` or `STRING`      | defaults to NUMERIC, otherwise, STRING    |
+| `String`           | `STRING`                   |                                           |
+| `Boolean`          | `BOOL`                     |                                           |
+| `Binary`           | `BYTES`                    |                                           |
+| `Null`             | A nullable column type     |                                           |
+| `List`             | `STRING`                   | json encoding                             |
+| `Map`              | `STRING`                   | json encoding                             |
+| `StringSet`        | `ARRAY<STRING>`            |                                           |
+| `NumberSet`        | `ARRAY<NUMERIC or STRING>` |                                           |
+| `BinarySet`        | `ARRAY<BYTES>`             |                                           |
+
+We discuss these, as well as other limits and notes on schema conversion, in the
+following sections.
+
+### Schema Inference
+
+DynamoDB is schemaless and it is impossible to get the full information of the
+schema. So we sample a set of rows and use them to model the schema in Cloud
+Spanner. For a large-scale table, scanning the entire table would be extremely
+slow. Therefore, we can only scan a part of it. We provide a flag
+`schema-sample-size` to specify how many rows to use for schema inference.
+
+In addition, for the performance purpose, we choose to call Scan on the table
+instead of randomly sample rows. This might be biased but we believe that the 
+result of Scan is not order-guaranteed and it can represent a sampled part
+of the table.
+
+### `Number`
+
+The Number type in DynamoDB is encoded as a string, which can represent an
+integer or float value. The range of the
+[Numeric type](https://cloud.google.com/spanner/docs/storing-numeric-data)
+in Cloud Spanner is smaller than the range of the
+[Number type](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.NamingRulesDataTypes.html)
+in DynamoDB. The conversion would result in out of range with potential
+precision loss. For simplicity, we will first try to convert it into a NUMERIC
+type. If it fails, we convert it into a STRING type. 
+
+### `Null` Data Type
+
+In DynamoDB, an attribute can have a Null data type that represents an unknown
```

Comment:
This is the first time you've used attribute.
Maybe use it earlier when talking about schema i.e. say something "DynamoDB attributes are mapped to Spanner columns" and then use that terminology consistently when referring the DynamoDB columns.
Or just use column consistently throughout (including here).